tests/e2e/test_pipelines_main_schema.py

```diff
@@ -120,10 +120,25 @@ def test_pipeline_splink_dummy(
         # Check that the results file matches the expected value
         results = (
             pd.read_parquet(final_output)
-            .sort_values("Input Record ID")
+            .sort_values(["Input Record Dataset", "Input Record ID"])
+            .reset_index(drop=True)
+        )
+        correct_results = (
+            pd.read_csv(correct_results_csv)
+            .sort_values(["Input Record Dataset", "Input Record ID"])
             .reset_index(drop=True)
         )
-        correct_results = pd.read_csv(correct_results_csv)
+
+        eq_df = correct_results.eq(results)
+        from functools import reduce
+
+        import numpy as np
+
+        eq_ser = reduce(np.logical_and, (eq_df[c] for c in eq_df.columns))
+        print("Printing any non-equal rows:")
+        print(results[~eq_ser])
+        print(correct_results[~eq_ser])
```

Comment:
It might be a matter of preference, but did you try `results.compare(correct_results, keep_equal=True, result_names=('actual', 'expected'))`? Could be a shorter way to print what you are getting at here.